endaq/calc/psd.py

```diff
@@ -236,6 +236,52 @@ def to_octave(
     return result
 
 
+def _aligned_bin_width(fstart: float = 1.0, octave_bins: float = 12.0) -> float:
+    """
+    Calculate a "good" linearly-spaced bin width, from a PSD of which an
+    octave-spaced PSD can be accurately calculated.
+
+    When computing the octave-spaced PSD, linear bins are mapped to octave bins
+    and the linear bin values are then summed together within each octave bin.
+    However, this calculation can produce an octave-spaced PSD that is an
+    *inaccurate* representation of the original linearly-spaced PSD, if the
+    mapping from linear to octave bins is *poorly aligned*; e.g., if a
+    significant linear bin is halfway between two adjacent octave bins.
+
+    Thus, the main criterion for a "good" linear spacing is how well the bin
+    borders in the linearly-spaced PSD align with those of the octave-spaced
+    PSD. This is mostly only significant at the lower frequencies, where the
+    octave-spaced bins comprise only a handful of linearly-spaced bins.
+
+    :param fstart: the starting frequency of the octave-spaced PSD
+    :param octave_bins: the number of bins per octave in the octave-spaced PSD
+    :return: a bin-width for a linearly-spaced PSD that aligns well with an
+        octave-spaced PSD defined by the given spacing parameters
+
+    .. todo::
+        allow user to adjust output resolution:
+        - add "good"-ness as a parameter to adjust resolution of result?
+        - iterate over solutions with increasing resolutions -> user just picks
+          the most appropriate one?
```

Comment:
Is this something we want rendered in the docs?

Reply:
this will be hidden in the docs [unless the config is set to show them](https://www.sphinx-doc.org/en/master/usage/extensions/todo.html#directive-todo), which we don't set in [our config](https://github.com/MideTechnology/endaq-python/blob/main/docs/conf.py)